core/src/main/java/org/apache/iceberg/SetSnapshotOperation.java

```diff
@@ -113,7 +113,7 @@ public void commit() {
         .run(taskOps -> {
           Snapshot snapshot = apply();
           TableMetadata updated = TableMetadata.buildFrom(base)
-              .setCurrentSnapshot(snapshot.snapshotId())
+              .setBranchSnapshot(snapshot.snapshotId(), SnapshotRef.MAIN_BRANCH)
```

Comment:
do we need to handle the same for rollback() also ?

Reply:
what do you mean by handle rollback? It's in this same class and committed here. The snapshot at L115 after `apply` is the one either rolled back or set by user.

Reply:
rollback can be branch specific operation right? 

As user will pass specific branch's snapshot id. We need a validation to check if the user passed snapshot is present in this branch or not ?

Reply:
Also here, we need to map to branch's snapshot ?

https://github.com/apache/iceberg/blob/4790e6c68eba7023c5792d33a4ad27d3971b2202/core/src/main/java/org/apache/iceberg/SetSnapshotOperation.java#L72

Reply:
That feature is not added yet. We are only keeping this the same as the behavior before introducing branching. We will start to allow custom branch after adding the related methods for branching and cherry-picking.

Reply:
ok. I can take up some work in this area. Will discuss offline.